doc/en/deprecations.rst

```diff
@@ -221,11 +221,32 @@ Using ``pytest.warns(None)``
 
 .. deprecated:: 7.0
 
-:func:`pytest.warns(None) <pytest.warns>` is now deprecated because many people used
-it to mean "this code does not emit warnings", but it actually had the effect of
-checking that the code emits at least one warning of any type - like ``pytest.warns()``
+:func:`pytest.warns(None) <pytest.warns>` is now deprecated because it was frequently misused.
+Its correct usage was checking that the code emits at least one warning of any type - like ``pytest.warns()``
 or ``pytest.warns(Warning)``.
 
+If you are looking to:
```

Comment:
I like this section and the examples a lot, so much I feel they should be part of the normal documentation, as opposed to just a note on the deprecations page.

How about a new section in `capture-warnings.rst`, pretty much what we have here? Then we can put a link to that section in here.